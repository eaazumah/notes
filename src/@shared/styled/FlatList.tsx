import {FlatList as _FlatList} from 'react-native';
import styled from 'styled-components/native';

const FlatList = styled.FlatList`
  background-color: ${props => props.theme.backgroundColor};
` as unknown as typeof _FlatList;

export default FlatList;
