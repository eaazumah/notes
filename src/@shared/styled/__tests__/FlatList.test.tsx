import 'jest-styled-components';
import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import TestProvider from '../../../__testUtils__/Providers';
import FlatList from '../FlatList';

test('styled Container works', () => {
  const tree = renderer
    .create(
      <TestProvider>
        <FlatList data={[]} renderItem={() => <Text />} />
      </TestProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
