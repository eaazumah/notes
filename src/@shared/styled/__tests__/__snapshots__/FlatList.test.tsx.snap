// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled Container works 1`] = `
<RCTScrollView
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "#F5F5F5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
